server/src/main/kotlin/ireader/core/prefs/PreferenceStore.kt

```diff
@@ -0,0 +1,220 @@
+package ireader.core.prefs
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.modules.SerializersModule
+import java.util.prefs.Preferences
+
+/**
+ * Implementation of IReader's PreferenceStore
+ * Must be in ireader.core.prefs package to match extension expectations
+ */
+class PreferenceStoreImpl(
+    private val packageName: String,
+) : PreferenceStore {
+    private val prefs = Preferences.userRoot().node("suwayomi/ireader/$packageName")
```

Comment:
Instead of storing in the JVM Preferences lets use our implementation of SharedPreferences in AndroidCompat. For example
```suggestion
    private val prefs = Injekt.get<Application>().getSharedPreferences("suwayomi/ireader/$packageName", Context.MODE_PRIVATE)
```